poetry/utils/helpers.py

```diff
@@ -29,17 +30,32 @@ def normalize_version(version):  # type: (str) -> str
 
 @contextmanager
 def temporary_directory(*args, **kwargs):
-    try:
-        from tempfile import TemporaryDirectory
+    name = tempfile.mkdtemp(*args, **kwargs)
 
-        with TemporaryDirectory(*args, **kwargs) as name:
-            yield name
-    except ImportError:
-        name = tempfile.mkdtemp(*args, **kwargs)
+    yield name
 
-        yield name
+    robust_rmtree(name)
 
-        shutil.rmtree(name)
+
+def robust_rmtree(path):
+    """Robustly tries to delete paths.
+
+    Retries several times if an OSError occurs.
+    If the final attempt fails, the Exception is propagated
+    to the caller.
+    """
+    timeout = 0.001
+    while timeout < 2:
```

Comment:
why 2 seconds instead 1 like the python version? Without a strong justification otherwise, I think we should use `1.0`

Reply:
I was running 2 seconds simply becuase I felt that it wasn't an unreasonable wait that little extra longer to increase robustness when this is simply gating a crash situation (i.e. the program will error at the end of the timeout if it fails).

CPython is using it in their tests, so obviously they are much more concious of the tradeoff between robustness and speed.

That being said, 1s is almost certainly enough so I'm fine switching it to that. And we can increase it in the future if people still face the issue.